### ISSUE TITLE
Radius marker for your own player

### ISSUE THREAD

**Issue by Alexgopen** [Skretzo/runelite-plugins#26]
I see that you can attach radius markers to npcs by NPC ID, but is there a way to attach a radius marker to your own player character?  

I want to be able to mark out a square radius around my player to mark out things such as NPC render distance, as well as the attack range of my weapon/spell.  I basically want a square around my character that updates with me as I run around.

**Comment by Skretzo** [comment]
You should try out the "Line of Sight" hub plugin, as that one will let you choose a radius around the player. It has an outline mode and will disregard tiles that you cannot walk on, which is probably useful for the attack range of a player weapon/spell.

I will eventually implement the ability to make a radius marker for your own player character, but that will not account for line of sight.

**Comment by Alexgopen** [comment]
Yeah I saw and tried that plugin before, hoping it would do what I was looking for, but it doesn't.  That one seems to be line of sight as in which obstacles will block your shot.  I'm looking to just have a square around the player which extends out to the edge of NPC render distance, to know if an area up ahead is close enough that a crystal impling would be visible to me if there were one there, that way I don't need to run up closer than I need to in order to check.  This would be one example of where it would be useful to have a square of predetermined size centered on the player, but there would be other cases such as avoiding being seen when doing wildy clue steps by keeping hotspots out of your render distance, among other use cases we haven't thought of yet.  

Just a simple square radius centered on the player

**Comment by Skretzo** [comment]
> In this case I have a 15 tile radius, as that's the distance at which NPCs begin rendering. Any NPCs outside this radius are not in render distance yet, so to search for crystal implings, I would run around covering as much area as I could with the marked radius. Then I don't need to waste energy going further than I need to if the area up ahead is already within the 15 tile NPC render distance

Have you tried lowering the render distance to 15 tiles in the GPU plugin? That way everything > 15 tiles away from the player should become black (or whatever colour you have set in the skybox plugin).

There is also a very scuffed workaround to do the marking in the Radius Marker plugin which is to bring a cat and have a marker follow the cat, although that will always be off by 1 tile and the cat will get stuck behind obstacles and need to be called frequently.

I will add the option to place radius markers that will follow the player, but I cannot give you an ETA for when this will happen. Hopefully within a month.
